core/tests/test_views.py

```diff
@@ -69,3 +69,111 @@ def test_add_post_unauthorized(client: Client, community: Community) -> None:
+
+
+def test_add_comment_invalid(client: Client, user: User, post: Post) -> None:
+    data = {"parent_id": post.pk, "content": ""}
+    client.force_login(user)
+    response = client.post(reverse("post-detail", kwargs={"pk": post.pk}), data=data)
+    assert response.status_code == 200
+    form = response.context["form"]
+    assert len(form.errors) == 1
+    assert "This field is required." in form.errors["content"]
+
+
+def test_add_comment_valid_special_characters(client: Client, user: User, post: Post) -> None:
+    data = {"parent_id": post.pk, "content": "This is a test comment with special characters! 😊🚀✨"}
+    client.force_login(user)
+    response = client.post(reverse("post-detail", kwargs={"pk": post.pk}), data=data, follow=True)
+    assert response.status_code == 200
+    new_comment = Post.objects.get(content=data["content"])
+    assert new_comment.content == data["content"]
+
+
+def test_add_comment_unauthorized(client: Client, post: Post) -> None:
+    data = {
+        "parent_id": post.pk,
+        "content": "This is a test comment content.",
+    }
+    response = client.post(reverse("post-detail", kwargs={"pk": post.pk}), data=data)
+    assert response.status_code == 302
+    assert reverse("login") in response.url
+
+
+def test_add_nested_comment_valid(client: Client, another_user: User, post: Post, comment: Post) -> None:
+    data = {
+        "parent_id": comment.pk,
+        "content": "This is a test nested comment content.",
+    }
+    client.force_login(another_user)
+    assert post.children_count == 1
+    assert post.get_comments().count() == 1
+    assert comment.children_count == 0
+    assert comment.get_comments().count() == 0
+    response = client.post(reverse("post-detail", kwargs={"pk": post.pk}), data=data, follow=True)
+    assert response.status_code == 200
+    post.refresh_from_db()
+    comment.refresh_from_db()
+    assert post.children_count == 2
+    assert post.get_comments().count() == 1
+    assert comment.children_count == 1
+    assert comment.get_comments().count() == 1
+
+    new_comment = Post.objects.get(content=data["content"])
+    assert new_comment.author == another_user
+    assert new_comment.parent == comment
+
+
+def test_add_nested_comment_invalid(client: Client, user: User, post: Post, comment: Post) -> None:
+    data = {"parent_id": comment.pk, "content": ""}
+    client.force_login(user)
+    response = client.post(reverse("post-detail", kwargs={"pk": post.pk}), data=data)
+    assert response.status_code == 200
+    form = response.context["form"]
+    assert len(form.errors) == 1
+    assert "This field is required." in form.errors["content"]
+
+
+def test_add_nested_comment_unauthorized(client: Client, post: Post, comment: Post) -> None:
+    data = {
+        "parent_id": comment.pk,
+        "content": "This is a test nested comment content.",
+    }
+    response = client.post(reverse("post-detail", kwargs={"pk": post.pk}), data=data)
```

Comment:
This can happen only in case of some kind of attack right? When someone is just sending post on the endpoint?
So basically this should not happen as first user would need to get CSRF token from the form page, which requires login. 
I am not saying it is wrong, I am just trying to think it through. Do you agree with my thought process, or, if I am wrong, can you explain?

Reply:
I agree